experimental/InvariantTheory/iterators.jl

```diff
@@ -311,3 +311,159 @@ function iterate_linear_algebra(BI::InvRingBasisIterator, state::Int)
   end
   return f, state + 1
 end
+
+################################################################################
+#
+#  "Iterate" a vector space
+#
+################################################################################
+
+function vector_space_iterator(K::FieldT, basis_iterator::IteratorT) where {FieldT <: Union{Nemo.GaloisField, Nemo.GaloisFmpzField, FqNmodFiniteField, FqFiniteField}, IteratorT}
+  return VectorSpaceIteratorFiniteField(K, basis_iterator)
+end
+
+vector_space_iterator(K::FieldT, basis_iterator::IteratorT, bound::Int = 10^5) where {FieldT, IteratorT} = VectorSpaceIteratorRand(K, basis_iterator, bound)
+
+Base.eltype(VSI::VectorSpaceIterator{FieldT, IteratorT, ElemT}) where {FieldT, IteratorT, ElemT} = ElemT
+
+Base.length(VSI::VectorSpaceIteratorFiniteField) = BigInt(order(VSI.field))^length(VSI.basis_iterator) - 1
+
+# The "generic" iterate for all subtypes of VectorSpaceIterator
+function _iterate(VSI::VectorSpaceIterator)
+  if isempty(VSI.basis_iterator)
+    return nothing
+  end
+
+  if length(VSI.basis_collected) > 0
+    b = VSI.basis_collected[1]
+  else
+    b, s = iterate(VSI.basis_iterator)
+    VSI.basis_collected = typeof(b)[ b ]
```

Comment:
I am pretty sure this is redundant
```suggestion
    VSI.basis_collected = [ b ]
```